.github/scripts/label_promoted_commits.py

```diff
@@ -23,36 +23,91 @@ def parser():
-    parser.add_argument('--label', type=str, required=True, help='Label to use')
+    parser.add_argument('--label', type=str, default='promoted-to-master', help='Label to use')
+    parser.add_argument('--ref', type=str, required=True, help='PR target branch')
     return parser.parse_args()
 
 
+def add_comment_and_close_pr(pr_number, repository, headers, comment):
+    comment_url = f'https://api.github.com/repos/{repository}/issues/{pr_number}/comments'
+    comment_data = {
+        "body": comment
+    }
+    response = requests.post(comment_url, headers=headers, json=comment_data)
+    if response.ok:
+        print(f'Comment added successfully to PR #{pr_number}')
+    else:
+        print(f'Failed to add comment to PR #{pr_number}. Status code: {response.status_code}')
+
+    close_url = f'https://api.github.com/repos/{repository}/pulls/{pr_number}'
+    close_data = {
+        "state": "closed"
+    }
+    response = requests.patch(close_url, headers=headers, json=close_data)
+    if response.ok:
+        print(f'PR #{pr_number} closed successfully')
+    else:
+        print(f'Failed to close PR #{pr_number}. Status code: {response.status_code}')
+
+
 def main():
     args = parser()
     pr_pattern = re.compile(r'Closes .*#([0-9]+)')
     g = Github(github_token)
     repo = g.get_repo(args.repository, lazy=False)
-
     commits = repo.compare(head=args.commit_after_merge, base=args.commit_before_merge)
+    processed_prs = set()
     # Print commit information
     for commit in commits.commits:
         print(commit.sha)
         match = pr_pattern.search(commit.commit.message)
         if match:
-            pr_number = match.group(1)
-            url = f'https://api.github.com/repos/{args.repository}/issues/{pr_number}/labels'
-            data = {
-                "labels": [f'{args.label}']
+            search_url = f'https://api.github.com/search/issues'
+            query = f"repo:{args.repository} is:pr is:closed sha:{commit.sha}"
+            params = {
+                "q": query,
             }
             headers = {
                 "Authorization": f"token {github_token}",
                 "Accept": "application/vnd.github.v3+json"
             }
-            response = requests.post(url, headers=headers, json=data)
-            if response.ok:
-                print(f"Label added successfully to {url}")
-            else:
-                print(f"No label was added to {url}")
+            response = requests.get(search_url, headers=headers, params=params)
+            prs = response.json().get("items", [])
+            for pr in prs:
+                refs_match = re.findall(r'Refs #(\d+)', pr["body"])
+                if refs_match:
+                    pr_number = int(refs_match[0])
+                    backport_pr_number = match.group(1)
+                    if pr_number in processed_prs:
+                        continue
+                    ref = re.search(r'-(\d+\.\d+)', args.ref)
```

Comment:
what if `github.ref` is not a branch named like `branch-5.3`, but something like `refs/heads/master` ?

Reply:
I thought that 
```
refs_match = re.findall(r'Refs #(\d+)', pr["body"])
if refs_match:
```
would prevent from this to happened, but there is a point, i will fix it